Removed buildtest build  option --disable-executor-check

Mariamajib opened the pull request.

Mariamajib committed:
removed option --disable-executor-check from buildtest/cli/__init__.py

Mariamajib committed:
removed option --disable-executor-check from bash_completion.sh

Mariamajib committed:
removed logic for validation_executors in buildtest/config.py

Mariamajib committed:
removed logic validation_executors in buildtest/main.py

Mariamajib committed:
removed variable validate_executors in  method configuration.validate in buildtest/main.py

szuananwar commented on buildtest/cli/__init__.py:
```diff
@@ -670,7 +670,6 @@ def build_menu(subparsers):
         type=positive_number,
     )
     extra_group.add_argument(
```
Please remove lines from 672-676 as we discussed.

Mariamajib replied:
@szuananwar I have removed lines from 672-676 as stated

szuananwar commented on bash_completion.sh:
```diff
@@ -162,7 +162,7 @@ _buildtest ()
     build|bd)
       local shortoption="-b -e -et -f -m -s -t -u -x -xt"
-      local longoption="--buildspec --disable-executor-check --executor --executor-type --exclude --exclude-tags --filter --helpfilter --limit --maxpendtime --modules --module-purge --nodes --pollinterval --procs --profile --rerun --remove-stagedir --retry --save-profile --stage --tags --timeout --unload-modules"
+      local longoption="--buildspec --executor --executor-type --exclude --exclude-tags --filter --helpfilter --limit --maxpendtime --modules --module-purge --nodes --pollinterval --procs --profile --rerun --remove-stagedir --retry --save-profile --stage --tags --timeout --unload-modules"
```
This is correct, no more deletion needed.

szuananwar commented on buildtest/config.py:
This is correct, no more deletion needed.

szuananwar commented on buildtest/main.py:
This is correct, no more deletion needed.

szuananwar reviewed (commented):
@Mariamajib Please see my comments.

Mariamajib committed:
Merge branch 'buildtesters:devel' into Mariam_patch

shahzebsiddiqui commented on buildtest/config.py:
```diff
@@ -119,7 +119,7 @@ def detect_system(self):
         # self.localexecutors = list(self.target_config["executors"]["local"].keys())
 
-    def validate(self, validate_executors=True, moduletool=None):
+    def validate(self, moduletool=None):
```
can you also update the docstring there is line below that has 

```
validate_executors (bool): Check executor settings. This is the default behavior but can be disabled
```

Mariamajib replied:
> can you also update the docstring there is line below that has
> 
> ```
> validate_executors (bool): Check executor settings. This is the default behavior but can be disabled
> ```

@shahzebsiddiqui I have updated the docstring

shahzebsiddiqui reviewed (changes requested):
please make this one change you should be all set

Mariamajib committed:
removed line 672-675 in buildtest/cli/__init__.py

Mariamajib committed:
Merge branch 'Mariam_patch' of https://github.com/Mariamajib/buildtest into Mariam_patch

Mariamajib committed:
updated the docstring in buildtest/config.py

Mariamajib committed:
updated the doc string in buildtest/config.py

Mariamajib committed:
removed validate method from buildtest/tools/tutorialexamples.py

Mariamajib committed:
updated scripts/spack_container/doc-examples.py

Mariamajib committed:
removed validate method from tests/executors/test_base.py

Mariamajib committed:
Merge branch 'devel' into Mariam_patch

Mariamajib committed:
updated scripts/spack_container/doc-examples.p

Mariamajib committed:
Merge branch 'Mariam_patch' of https://github.com/Mariamajib/buildtest into Mariam_patch

Mariamajib committed:
Updated the validate method in buildtest/tools/tutorialexamples.py

Mariamajib committed:
updated validate method in tests/executors/test_base.py

Mariamajib committed:
updated the --init.py

shahzebsiddiqui committed:
we need to invoke method _executor_check in SiteConfiguration class inorder to build
a list of valid executors. This was not invoked which caused regression test to fail because buildspecs
were not validated properly

Mariamajib committed:
1. Updated _executor_check method in config.py
2. Made a comment in test_builders.py about a bug fixed
3. Moved test_stats.py from ../tests/cli to tests

Mariamajib committed:
Merge remote-tracking branch 'origin/Mariam_patch' into Mariam_patch

Mariamajib committed:
updated test_builders.py

Mariamajib committed:
updated test_stats.py
